src/Api/Vault/AuthorizationHandlers/Collections/BulkCollectionAuthorizationHandler.cs

```diff
@@ -0,0 +1,205 @@
+        ICollection<Collection> targetCollections, CurrentContextOrganization org)
+    {
+        // Owners, Admins, Providers, and users with DeleteAnyCollection permission can always delete collections
+        if (
+            org.Type is OrganizationUserType.Owner or OrganizationUserType.Admin ||
+            org.Permissions is { DeleteAnyCollection: true } ||
+            await _currentContext.ProviderUserForOrgAsync(org.Id))
+        {
+            context.Succeed(requirement);
+            return;
+        }
+
+        // The limit collection management setting is enabled and we are not an Admin (above condition), fail
+        if (org.LimitCollectionCreationDeletion)
+        {
+            context.Fail();
+            return;
+        }
+
+        await CheckCollectionPermissionsAsync(context, requirement, targetCollections, org, requireManagePermission: true);
+    }
+
+    /// <summary>
+    /// Ensures the acting user is allowed to manage access permissions for the target collections.
+    /// </summary>
+    private async Task CanManageCollectionAccessAsync(AuthorizationHandlerContext context,
+        IAuthorizationRequirement requirement, ICollection<Collection> targetCollections, CurrentContextOrganization org)
+    {
+        // Owners, Admins, Providers, and users with EditAnyCollection permission can always manage collection access
+        if (
+            org.Permissions is { EditAnyCollection: true } ||
+            org.Type is OrganizationUserType.Owner or OrganizationUserType.Admin ||
+            await _currentContext.ProviderUserForOrgAsync(org.Id))
+        {
+            context.Succeed(requirement);
+            return;
+        }
+
+        await CheckCollectionPermissionsAsync(context, requirement, targetCollections, org, requireManagePermission: true);
+    }
+
+    private async Task CheckCollectionPermissionsAsync(
+        AuthorizationHandlerContext context,
+        IAuthorizationRequirement requirement,
+        ICollection<Collection> targetCollections,
+        CurrentContextOrganization org,
+        bool requireManagePermission)
+    {
+        // List of collection Ids the acting user has access to
+        var manageableCollectionIds =
+            (await _collectionRepository.GetManyByUserIdAsync(_currentContext.UserId!.Value))
+            .Where(c =>
+                // If requireManagePermission is true, check Collections with Manage permission
+                (!requireManagePermission || c.Manage)
+                && c.OrganizationId == org.Id)
+            .Select(c => c.Id)
+            .ToHashSet();
+
+        // The acting user does not have permissions for all target collections, fail
+        if (targetCollections.Any(tc => !manageableCollectionIds.Contains(tc.Id)))
+        {
+            context.Fail();
+            return;
+        }
+
+        context.Succeed(requirement);
```

Comment:
I recommend inverting this logic to state the positive/affirmative case (assuming we're returning a bool):

```suggestion
  return targetCollections.All(tc => manageableCollectionIds.Contains(tc.Id));
```